### PR TITLE
Fixing pad op lowering pipeline from subtree merge

### PR DESCRIPTION
**Pull request by jerryyin**
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/521

### CODE REVIEW

File: mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp [hunk 1]
```diff
@@ -46,6 +46,10 @@ void miopen::addHighLevelPipeline(PassManager &pm, bool toMIOpen) {
     pm.addPass(tosa::createTosaToMIOpenPass());
   }
   mlir::tosa::addTosaToLinalgPasses(pm);
+  bufferization::OneShotBufferizationOptions options;
+  options.allowReturnAllocs = true;
+  options.testAnalysisOnly = true;
```
Comment (sjw36) [inline mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp:51]: Why is testAnalysisOnly necessary. From the description it seems to suggest that it won't do any of the transforms.

File: mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp [hunk 2]
```diff
@@ -46,6 +46,10 @@ void miopen::addHighLevelPipeline(PassManager &pm, bool toMIOpen) {
...
+  pm.addPass(createLinalgComprehensiveModuleBufferizePass(options));
   pm.addPass(createLinalgElementwiseOpFusionPass());
   pm.addPass(createLinalgBufferizePass());
```
Comment (sjw36) [inline mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp:54]: Many of these will not become unnecessary. Only bufferization::createBufferResultsToOutParamsPass is still needed for Bufferization.

**Review by sjw36 — APPROVED** [review]
Overall looks good but there is some redundancy now.

File: mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp (hunk 2 above)
Comment (jerryyin) [reply mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp:54]: They are actually still useful. I did a binary search in attempting to removing the rest of those passes and they will result in unit test failures one way or the other. For the most part, I think comprehensive module bufferize pass is still orthogonal to those other passes.

File: mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp (hunk 1 above)
Comment (jerryyin) [reply mlir/lib/Dialect/MIOpen/Pipeline/Pipeline.cpp:51]: By looking at their pass pipeline:

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/579526e08448ce42d89528f75c870175c337c34b/external/llvm-project/mlir/lib/Dialect/Linalg/Transforms/ComprehensiveBufferizePass.cpp#L117-L124

Everything below `opt.testAnalysisOnly` will break MIOpen dialect and the reason is not yet clear to me. But I can confirm that everything before the escape is equivalent to previous pipeline.

**Pull request merged**